### ISSUE TITLE
LaTeX .sty files missing

### ISSUE THREAD

**Issue by christetreault** [ucsd-progsys/liquidhaskell-tutorial#16]
When building the book, various `.sty` files are missing. I've been downloading them one-by-one until `make` stops failing, but I've downloaded 11 so far, and there is no end in sight.

Is there a list of LaTeX packages required for the book to build? Is there some sort of combined package or distribution that I can download that has all the dependencies? Failing that, can we get the `.sty` files added to the repository (barring licensing issues), so that we can just call `make` and have it work?

**Comment by ranjitjhala** [comment]
Can you name some of them? there should be one -- like 'texlive-extras' or
some such...

but maybe its easier just to read the HTML version for now? (could be
easier to fix this problem in person?)

On Fri, Aug 7, 2015 at 3:33 AM, Chris Tetreault <notifications@github.com>
wrote:

> When building the book, various .sty files are missing. I've been
> downloading them one-by-one until make stops failing, but I've downloaded
> 11 so far, and there is no end in sight.
>
> Is there a list of LaTeX packages required for the book to build? Is there
> some sort of combined package or distribution that I can download that has
> all the dependencies? Failing that, can we get the .sty files added to
> the repository (barring licensing issues), so that we can just call make
> and have it work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell-tutorial/issues/16>.
>



-- 
Ranjit.

**Comment by christetreault** [comment]
I was working on adding a "building the book" section to the readme. I figure it would be good to have in there since we've had so many discussions about it already.

Additionally, the .pdf is in the repo. But if I'm going to make formatting changes and such, I'd like to be able to build the updates and make sure everything looks right.

The following files are the ones I got through before I got fed up:

* xifthen.sty
* imftarg.sty
* titlesec.sty
* titletoc.sty
* chngpage.sty
* paralist.sty
* optparams.sty
* placeins.sty
* comment.sty
* thmtools.sty
* thmpatch.sty

If it's just a texlive addon that has this stuff I'll see if I can't figure out which one. CTAN kept claiming that these were in texlive.

**Comment by ranjitjhala** [comment]
Yes, look for texlive and maybe texlive-extras.

also don't install them one by one -- usually there is a gigantic ubuntu
package
with the lot (I think texlive or texlive extra or some such.)



On Fri, Aug 7, 2015 at 8:55 PM, Chris Tetreault <notifications@github.com>
wrote:

> I was working on adding a "building the book" section to the readme. I
> figure it would be good to have in there since we've had so many
> discussions about it already.
>
> Additionally, the .pdf is in the repo. But if I'm going to make formatting
> changes and such, I'd like to be able to build the updates and make sure
> everything looks right.
>
> The following files are the ones I got through before I got fed up:
>
>    - xifthen.sty
>    - imftarg.sty
>    - titlesec.sty
>    - titletoc.sty
>    - chngpage.sty
>    - paralist.sty
>    - optparams.sty
>    - placeins.sty
>    - comment.sty
>    - thmtools.sty
>    - thmpatch.sty
>
> If it's just a texlive addon that has this stuff I'll see if I can't
> figure out which one. CTAN kept claiming that these were in texlive.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell-tutorial/issues/16#issuecomment-128732888>
> .
>



-- 
Ranjit.

**Comment by ranjitjhala** [comment]
texlive-latex and texlive-latex extra.

On Fri, Aug 7, 2015 at 9:17 PM, Ranjit Jhala <jhala@cs.ucsd.edu> wrote:

> Yes, look for texlive and maybe texlive-extras.
>
> also don't install them one by one -- usually there is a gigantic ubuntu
> package
> with the lot (I think texlive or texlive extra or some such.)
>
>
>
> On Fri, Aug 7, 2015 at 8:55 PM, Chris Tetreault <notifications@github.com>
> wrote:
>
>> I was working on adding a "building the book" section to the readme. I
>> figure it would be good to have in there since we've had so many
>> discussions about it already.
>>
>> Additionally, the .pdf is in the repo. But if I'm going to make
>> formatting changes and such, I'd like to be able to build the updates and
>> make sure everything looks right.
>>
>> The following files are the ones I got through before I got fed up:
>>
>>    - xifthen.sty
>>    - imftarg.sty
>>    - titlesec.sty
>>    - titletoc.sty
>>    - chngpage.sty
>>    - paralist.sty
>>    - optparams.sty
>>    - placeins.sty
>>    - comment.sty
>>    - thmtools.sty
>>    - thmpatch.sty
>>
>> If it's just a texlive addon that has this stuff I'll see if I can't
>> figure out which one. CTAN kept claiming that these were in texlive.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ucsd-progsys/liquidhaskell-tutorial/issues/16#issuecomment-128732888>
>> .
>>
>
>
>
> --
> Ranjit.
>



-- 
Ranjit.

**Comment by christetreault** [comment]
The following two packages seem to get me through the "missing .sty file" section:

* texlive-latex-extra
* texlive-fonts-extra

Now I'm getting an error:

```
! Undefined control sequence.
<recently read> \tightlist 
                           
l.456 \tightlist
```

I feel like this might be something we'd need to work out in person. From looking at the Makefile (lines 1 and 2 in particular) I feel like the hidden dependency might be `RanjitsLaptop`.

This is certainly not an emergency though. I'll just assume my formatting changes don't look strange for the time being.

**Comment by ranjitjhala** [comment]
hmmm. wonder what thats about, can you see this:

https://github.com/rstudio/rmarkdown/issues/439

On Fri, Aug 7, 2015 at 9:40 PM, Chris Tetreault <notifications@github.com>
wrote:

> The following two packages seem to get me through the "missing .sty file"
> section:
>
>    - texlive-latex-extra
>    - texlive-fonts-extra
>
> Now I'm getting an error:
>
> ! Undefined control sequence.
> <recently read> \tightlist
>
> l.456 \tightlist
>
> I feel like this might be something we'd need to work out in person. From
> looking at the Makefile (lines 1 and 2 in particular) I feel like the
> hidden dependency might be RanjitsLaptop.
>
> This is certainly not an emergency though. I'll just assume my formatting
> changes don't look strange for the time being.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell-tutorial/issues/16#issuecomment-128749069>
> .
>



-- 
Ranjit.

**Comment by ranjitjhala** [comment]
actually, looks like this is more relevant:

http://tex.stackexchange.com/questions/257418/error-tightlist-converting-md-file-into-pdf-using-pandoc

sigh. latex & pandoc!



On Fri, Aug 7, 2015 at 10:07 PM, Ranjit Jhala <jhala@cs.ucsd.edu> wrote:

> hmmm. wonder what thats about, can you see this:
>
> https://github.com/rstudio/rmarkdown/issues/439
>
> On Fri, Aug 7, 2015 at 9:40 PM, Chris Tetreault <notifications@github.com>
> wrote:
>
>> The following two packages seem to get me through the "missing .sty file"
>> section:
>>
>>    - texlive-latex-extra
>>    - texlive-fonts-extra
>>
>> Now I'm getting an error:
>>
>> ! Undefined control sequence.
>> <recently read> \tightlist
>>
>> l.456 \tightlist
>>
>> I feel like this might be something we'd need to work out in person. From
>> looking at the Makefile (lines 1 and 2 in particular) I feel like the
>> hidden dependency might be RanjitsLaptop.
>>
>> This is certainly not an emergency though. I'll just assume my formatting
>> changes don't look strange for the time being.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ucsd-progsys/liquidhaskell-tutorial/issues/16#issuecomment-128749069>
>> .
>>
>
>
>
> --
> Ranjit.
>



-- 
Ranjit.

**Comment by christetreault** [comment]
I made the change locally and it builds now. I can go ahead and submit a pull request for this.

**Comment by christetreault** [comment]
OK, pull request opened (#17). After it gets merged, I vote this issue gets closed.

**Comment by ranjitjhala** [comment]
Fixed by #17
